pallets/parachain-staking/src/lib.rs

```diff
@@ -2768,59 +2786,46 @@ pub mod pallet {
 			inner.try_into().expect("Did not extend vec q.e.d.")
 		}
 
-		fn peaq_reward_mechanism_impl() {
-			let mut reads = Weight::from_parts(0, 1);
-			let mut writes = Weight::from_parts(0, 1);
+		/// Get a unique, inaccessible account id from the `PotId`.
+		pub fn account_id() -> T::AccountId {
+			T::PotId::get().into_account_truncating()
+		}
 
+		/// Handles staking reward payout for previous session for one collator and their delegators
+		fn payout_collator() {
 			let pot = Self::account_id();
-			let issue_number = T::Currency::free_balance(&pot)
-				.checked_sub(&T::Currency::minimum_balance())
-				.unwrap_or_else(Zero::zero);
-
-			let (in_reads, total_staking_in_session) = Self::get_total_collator_staking_num();
-			reads.saturating_add(in_reads);
+			// get payout info for the last round
+			let payout_info = DelayedPayoutInfo::<T>::get();
```

Comment:
Do we assume we only have one reward set to be distributed?

One weird scenario comes into my mind. When we do the runtime upgrade, that's possible the runtime upgrade at the 5 blocks before the session change. If that happened, could we handle it?

Reply:
Yup ~ In one session, we'll distribute to collators/delegators set of previous session only.
For runtime upgrade, at block 5 of a session for example, we'll distribute rewards as we always have, based on the state of stake at the end of a session, in the next session, our new mechanism will kick in. You're right about it being a weird case, if our new mechanism kicks in mid session, we'll have alot of problems.

Reply:
After discussion, we'll try to use the flag/force the new session in the migration. Therefore, we don't need to handle that